### PR TITLE
Set enable network host to false during integration tests.

### PR DESCRIPTION
**Pull request by Kaushikdkrikhanu**
**Short description of the problem this fixes or functionality that this introduces. This may be used for the CHANGELOG**

Temporary fix to this issue:https://github.com/All-Hands-AI/OpenHands/issues/3568

---
**Give a summary of what the PR does, explaining any non-trivial design decisions**

Currently in wsl, when network_host is set to 'host', the backend is unable to access the runitime server running inside docker. So, a permanent fix will be fixing the wsl to docker connection. But maybe not using the host network stack might just be safer. 

---
**Link of any specific issues this addresses**

### CODE REVIEW

**Review by xingyaoww — APPROVED** [review]
Thanks!

**Comment by xingyaoww** [comment]
ugh the integration tests failed bc we need to map 8000 port into the container... now i get it why we set it to host network before..

File: tests/integration/test_agent.py [hunk 1]
```diff
@@ -29,7 +29,7 @@
     workspace_base=os.getenv('WORKSPACE_BASE'),
     workspace_mount_path=os.getenv('WORKSPACE_MOUNT_PATH'),
     sandbox=SandboxConfig(
-        use_host_network=True,
+        use_host_network=False,
```
Comment (tobitege) [inline tests/integration/test_agent.py:32]: What if we remove this line instead, shouldn't an env or config.toml value be picked up then, iirc? 🤔 
And for GitHub CI we then can set the env var in the workflow.

Comment (rbren) [reply tests/integration/test_agent.py:32]: I think it's better to make the tests deterministic here--everyone should get the same result regardless of what's in their config.toml

**Review by rbren — APPROVED** [review]

File: tests/integration/test_agent.py (hunk 1 above)
Comment (tobitege) [reply tests/integration/test_agent.py:32]: It's not about that, it's about different platforms in combination with docker can result differently based on this setting. If turned off, I fear that these will break in WSL, for example.

**Comment by Kaushikdkrikhanu** [comment]
is anyone getting permission error when running this: TEST_RUNTIME=eventstream ONLY_TEST_NAME=test_ipython  ONLY_TEST_AGENT=CodeActAgent ./tests/integration/regenerate.sh 
image

**Comment by Kaushikdkrikhanu** [comment]
> ugh the integration tests failed bc we need to map 8000 port into the container... now i get it why we set it to host network before..

Which service is using port 8000 inside docker?

**Comment by tobitege** [comment]
Err... seeing this again, I'm running integration tests just fine in WSL with host network and don't see a reason to turn it off, tbh.

**Comment by tobitege** [comment]
> is anyone getting permission error when running this: TEST_RUNTIME=eventstream ONLY_TEST_NAME=test_ipython ONLY_TEST_AGENT=CodeActAgent ./tests/integration/

The permission error should be gone with 0.9.4 since we made sure that the user with id 1000 is removed from the base image.

File: tests/integration/test_agent.py (hunk 1 above)
Comment (tobitege) [reply tests/integration/test_agent.py:32]: But isn't this outside of the scope of the tests, as to me it is a platform issue and is not part of the test results.
The user should have the option to run the tests with the option being True or False.

**Comment by Kaushikdkrikhanu** [comment]
I am not working on this currently, closing this.

**Pull request closed without merging**